### PR TITLE
Make pubsub_last_item table swappable RDBMS backends

### PR DESCRIPTION
**Pull request by aleklisi**
Proposed changes include pubsub_last_item table with swappable RDBMS backends.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[5900.1](https://travis-ci.org/esl/MongooseIM/jobs/457936227) / Erlang 19.3 / small_tests / 86c687dc5ba5ca75e28c6f45f0ebb380b2d06e58
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2139/5900/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2139/5900/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-11-21_13.23.17/lib.mongooseim.logs/run.2018-11-21_13.23.18/suite.log.html)

**Comment by mongoose-im** [comment]
[5903.1](https://travis-ci.org/esl/MongooseIM/jobs/458416898) / Erlang 19.3 / small_tests / 3aa19154842a2cd14d16d63b6549d98e51e32e60
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2139/5903/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2139/5903/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-11-22_13.53.53/lib.mongooseim.logs/run.2018-11-22_13.53.54/suite.log.html)

**Comment by mongoose-im** [comment]
[5904.1](https://travis-ci.org/esl/MongooseIM/jobs/458426746) / Erlang 19.3 / small_tests / 85d94ad1b6a0b913c38f03d55fde378a299614ae
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2139/5904/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2139/5904/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-11-22_14.27.20/lib.mongooseim.logs/run.2018-11-22_14.27.21/suite.log.html)

File: src/pubsub/mod_pubsub.erl [hunk 1]
```diff
@@ -297,24 +297,24 @@ init([ServerHost, Opts]) ->
...
+        DB -> pubsub_cache:create_table(DB)
+    end.
+    
 store_config_in_ets(Host, ServerHost, Opts, Plugins, NodeTree, PepMapping) ->
     Access = gen_mod:get_opt(access_createnode, Opts, fun(A) when is_atom(A) -> A end, all),
     PepOffline = gen_mod:get_opt(ignore_pep_from_offline, Opts,
                                  fun(A) when is_boolean(A) -> A end, true),
     LastItemCache = gen_mod:get_opt(last_item_cache, Opts,
-                                    fun(A) when is_boolean(A) -> A end, false),
+                                    fun(A) when A == false orelse A == rdbms orelse A == mnesia -> A end, false),
```
Comment (aleklisi) [inline src/pubsub/mod_pubsub.erl]: `A == false orelse A == rdbms orelse A == mnesia`
to 
`A == rdbms orelse A == mnesia`

**Comment by mongoose-im** [comment]
[5905.1](https://travis-ci.org/esl/MongooseIM/jobs/458734441) / Erlang 19.3 / small_tests / a960a26dcec4a91a2e0be681d058ba6c85e638a2
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2139/5905/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2139/5905/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-11-23_10.11.46/lib.mongooseim.logs/run.2018-11-23_10.11.47/suite.log.html)

File: big_tests/tests/pubsub_SUITE.erl [hunk 2]
```diff
@@ -229,6 +247,18 @@ end_per_suite(Config) ->
     dynamic_modules:restore_modules(domain(), Config),
     escalus:end_per_suite(Config).
 
+init_per_group(pubsub_cache_with_mnesia, Config) ->
+    Args = proplists:get_value(mod_pubsub, required_modules()),
+    Args0 = [{last_item_cache, mnesia} | Args],
+    dynamic_modules:restart(domain(), mod_pubsub, Args0),
+    Config;
+
+init_per_group(pubsub_cache_with_rdbms, Config) ->
```
Comment (aleklisi) [inline big_tests/tests/pubsub_SUITE.erl]: note: most likely cleanup per group required

File: big_tests/tests/pubsub_SUITE.erl [hunk 3]
```diff
@@ -229,6 +247,18 @@ end_per_suite(Config) ->
     dynamic_modules:restore_modules(domain(), Config),
     escalus:end_per_suite(Config).
 
+init_per_group(pubsub_cache_with_mnesia, Config) ->
```
Comment (aleklisi) [inline big_tests/tests/pubsub_SUITE.erl]: note: most likely cleanup per group required

**Comment by mongoose-im** [comment]
[5918.1](https://travis-ci.org/esl/MongooseIM/jobs/460230669) / Erlang 19.3 / small_tests / c85c248b4caa7ac09ef6c42aa7ad93ddd699906c
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2139/5918/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2139/5918/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-11-27_12.49.40/lib.mongooseim.logs/run.2018-11-27_12.49.41/suite.log.html)

**Comment by mongoose-im** [comment]
[5921.1](https://travis-ci.org/esl/MongooseIM/jobs/460258308) / Erlang 19.3 / small_tests / 05ce37adbf48389ff296cb51f59354ab30502e3e
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2139/5921/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2139/5921/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-11-27_13.51.42/lib.mongooseim.logs/run.2018-11-27_13.51.43/suite.log.html)

**Pull request closed without merging**